#! /bin/sh /usr/share/dpatch/dpatch-run
## 132a391e-fix-glib-version-check.dpatch by  <anders@0x63.nu>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Do our own GLib version check instead of using glib_check_version, 
## DP: backported from xmms2-devel 132a391eae67fb586816341d92b7ca31600d824e

@DPATCH@

132a391eae67fb586816341d92b7ca31600d824e
diff --git a/src/xmms/main.c b/src/xmms/main.c
index c89ccbd..2bdaee0 100644
--- a/src/xmms/main.c
+++ b/src/xmms/main.c
@@ -365,9 +365,6 @@ main (int argc, char **argv)
 	int loglevel = 1;
 	xmms_playlist_t *playlist;
 	gchar default_path[XMMS_PATH_MAX + 16], *tmp;
-
-	const gchar *vererr;
-
 	gboolean verbose = FALSE;
 	gboolean quiet = FALSE;
 	gboolean version = FALSE;
@@ -398,9 +395,13 @@ main (int argc, char **argv)
 	};
 
 	/** Check that we are running against the correct glib version */
-	vererr = glib_check_version (GLIB_MAJOR_VERSION, GLIB_MINOR_VERSION, 0);
-	if (vererr) {
-		g_print ("Bad glib version: %s\n", vererr);
+	if (glib_major_version != GLIB_MAJOR_VERSION ||
+	    glib_minor_version < GLIB_MINOR_VERSION) {
+		g_print ("xmms2d is build against version %d.%d,\n"
+		         "but is (runtime) linked against %d.%d.\n"
+		         "Refusing to start.\n",
+		         GLIB_MAJOR_VERSION, GLIB_MINOR_VERSION,
+		         glib_major_version, glib_minor_version);
 		exit (EXIT_FAILURE);
 	}
 
